source/code-snippets/typescript/dot-notation.ts

```diff
@@ -0,0 +1,31 @@
+// start-no-error
+interface TestType {
+  field: {nested : number}; 
+}
+const database = client.db("<your db>");
+const collection = database.collection<TestType>("<your collection>");
+await collection.updateOne(
+  {},
+  {$set: { "field.nested" : "A string"}}
+);
+// end-no-error
+// start-error
+interface TestType {
+  field: {nested : number}; 
+}
+const database = client.db("<your db>");
+const collection = database.collection<TestType>("<your collection>");
+await collection.updateOne(
+  {},
+  {$set: { field : {nested : "A string"}}}
+);
+// end-error
+// start-no-key
+interface TestNumber {
+  myNumber : number
+}
+
+const database = client.db("<your db>");
```

Comment:
I think we should use consistent quotes (either all single or all double) for strings, I see both in this file

Reply:
Apologies, I forgot to run all code snippets through a formatter. Ran snippets through prettier formatter.